### PR TITLE
Make deploy/upgrade (to) a stable release possible

### PR DESCRIPTION
**Pull request by berendt**
Signed-off-by: Christian Berendt <berendt@osism.tech>

### CODE REVIEW

**Comment by berendt** [comment]
On the target it is required tu run MANAGER_VERSION={{ version_manager }} gilt overlay (twice) to update environments/manager/images.yml

**Comment by berendt** [comment]
Update prepare. Will push it tomorrow.

File: doc/source/deployment.rst [hunk 1]
```diff
@@ -34,3 +34,37 @@ The ``Run part 3`` task takes some time to complete, depending on the cloud. Run
 
    Path to the ``ansible-galaxy`` binary or the ``ansible-playbook`` only needs to be
    adjusted if the binaries are not findable via ``PATH``.
+
+
+Customise versions
+==================
+
+By default, the latest manager service, Ceph Pacific and OpenStack Yoga are deployed.
+This can be customises via the parameters ``version_ceph``, ``version_manager``, and
```
Comment (osfrickler) [inline doc/source/deployment.rst:43]: customised

File: doc/source/deployment.rst [hunk 2]
```diff
@@ -34,3 +34,37 @@ The ``Run part 3`` task takes some time to complete, depending on the cloud. Run
...
+
+.. code-block:: console
+
+   -e version_manager=4.0.0a
+
+.. note::
+
+   If a specific version of the manager and thus OSISM itself, a so-called stable release,
+   is deployed, the explicit specification of the Ceph version and the OpenStack version
+   is not necessary. The versions of Ceph and OpenStack are then determined from the stable
```
Comment (osfrickler) [inline doc/source/deployment.rst:68]: "not necessary" or even "not possible"?

File: scripts/deploy.sh [hunk 3]
```diff
@@ -12,6 +12,15 @@ source /opt/manager-vars.sh
 /opt/configuration/scripts/set-manager-version.sh $MANAGER_VERSION
 /opt/configuration/scripts/set-openstack-version.sh $OPENSTACK_VERSION
```
Comment (osfrickler) [inline scripts/deploy.sh:13]: Move set-ceph/set-openstack into an else block below? Avoids unnecessary change and confusion

File: scripts/upgrade.sh [hunk 4]
```diff
@@ -14,6 +14,15 @@ OPENSTACK_VERSION=${3:-yoga}
 /opt/configuration/scripts/set-manager-version.sh $MANAGER_VERSION
 /opt/configuration/scripts/set-openstack-version.sh $OPENSTACK_VERSION
```
Comment (osfrickler) [inline scripts/upgrade.sh:15]: dito

File: scripts/upgrade.sh [hunk 5]
```diff
@@ -14,6 +14,15 @@ OPENSTACK_VERSION=${3:-yoga}
...
 
+# NOTE: For a stable release, the versions of Ceph and OpenStack to use
+#       are set by the version of the stable release (set via the
+#       manager_version parameter) and not by release names.
+
+if [[ $MANAGER_VERSION != "latest" ]]; then
+    sed -i '/ceph_version:/d' /opt/configuration/environments/manager/configuration.yml
+    sed -i '/openstack_version:/d' /opt/configuration/environments/manager/configuration.yml
```
Comment (osfrickler) [inline scripts/upgrade.sh:23]: These need to be readded when upgrading from a stable version to latest

**Comment by berendt** [comment]
Failed because of a new back port on stable. Fixed with https://github.com/osism/ansible-defaults/pull/95.
